.github/codeowners/who_owns.py

```
@@ -0,0 +1,156 @@
+    for r in filetype_rules:
+        pat = r.get("pattern", "")
+        if pat and match(pat, filepath):
+            teams.update(r.get("request_review_from", []))
+    return teams
+
+
+def changed_files(repo: str, base: str) -> list[str]:
+    """Files changed vs ``base`` (merge-base diff), falling back to plain diff."""
+    for args in ([f"{base}...HEAD"], [base], []):
+        try:
+            out = subprocess.check_output(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)